subprojects/language-java/src/main/java/org/gradle/api/tasks/JavaExec.java

```diff
@@ -142,7 +147,7 @@ public JavaExec() {
     @TaskAction
     public void exec() {
         validateExecutableMatchesToolchain();
-        setJvmArgs(getJvmArgs()); // convention mapping for 'jvmArgs'
```

Comment:
I think we're going to have to do something to manually wire in the convention mapping again in case it's being used outside of our plugins.

Due to history, there are plugins in the wild that are using convention mapping and will break if we're no longer calling `getJvmArgs` somewhere.  Spring Boot seems to be one:
https://github.com/spring-projects/spring-boot/blob/57a4eb88dbef7c1cf25eaa27baaf21cdd0e45083/spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/java/org/springframework/boot/gradle/plugin/JavaPluginAction.java#L177

Reply:
@big-guy Yeah, I've also faced with issue like
`The value for task ':bootRun' property 'jvmArguments' is final and cannot be changed any further.`
Seems like this is because of Spring configuring `jvmArgs` in execution time
https://github.com/spring-projects/spring-boot/blob/3.0.x/spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/java/org/springframework/boot/gradle/tasks/run/BootRun.java#L68

I'm wondering how I can handle it?

Reply:
@gradle/bt-jvm WDYT? It seems Spring team will have to make a fixes at their side for stop configuring `jvmArgs` at execution time? And then we probably can ask them to get rid off `conventionMapping` usage also?
I'm not sure about other plugin authors tho

Reply:
Hrm.  `@Internal` seems like it would work for the problem with finalization.

Spring will need to change eventually, but we can't break builds and rely on the plugin being fixed. 

A couple of ideas...
- Use `ConventionMapping.getConventionValue` to grab the convention and set it on the new `ListProperty` in the constructor.
- Change `ConventionAwareHelper.map(...)` to know that `map("jvmArgs", <convention>)` should be `jvmArguments.convention(<convention>)`.  We have `@ReplacedBy` that's only used for documentation that would tell us the right property name.

Reply:
@lptr this is the PR I was talking about today